packages/driver/cypress/integration/commands/assertions_spec.js

```diff
@@ -127,6 +127,17 @@ describe('src/cy/commands/assertions', () => {
       }).should('deep.eq', { foo: 'baz' })
     })
 
+    // https://github.com/cypress-io/cypress/issues/16006
+    it(`does not show more than one .should('contain') assertion when chained after .should('be.visible')`, function () {
+      cy.get('#data-number')
+      .should('be.visible')
+      .should('contain', 'span')
+      .should('contain', 'with')
+      .then(function () {
+        expect(this.logs[2].get('message')).to.contain('**span**')
```

Comment:
Maybe this is already tested elsewhere. 

```suggestion
        expect(this.logs[2].get('message')).to.contain('**span**')
        expect(this.logs[3].get('message')).to.contain('**with**')
```